components/oss/types.go

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2021 Layotto Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package oss
+
+import (
+	"encoding/json"
+	"errors"
+)
+
+var (
+	ErrInvalid = errors.New("invalid argument")
+)
+
+// OssConfig wraps configuration for a oss implementation
+type OssConfig struct {
+	Metadata json.RawMessage `json:"metadata"`
```

Comment:
Can we change it to:
```
Metadata map[string]string `json:"metadata"`
```

We can't implement features like configuration injection if it's `json.RawMessage`

Reply:
users config may be like:

`
"oss": {
  "metadata": {
    "config1": {
    	"hello":"world"
    },
    "config2": {
    	"secret1":{

    	}
    	"key1":"value1",
    	"key2":"value2",
    }
  }
 }

`

may be can add an dynamicConf which type is Metadata map[string]string `json:"metadata"`, this metadata is static configure for component

Reply:
How about changing the type of metadata to `map[string]json.RawMessage`?

Reply:
done